File: Vanilla/sources/zdoc/media/lua/shared/Library/EventSound.lua
---@class EventSound : zombie.radio.StorySounds.EventSound
---@field protected name String
---@field protected color Color
---@field protected dataPoints ArrayList|Unknown
---@field protected storySounds ArrayList|Unknown
EventSound = {}

---@public
---@param arg0 ArrayList|Unknown
---@return void
function EventSound:setStorySounds(arg0) end

---@public
---@return ArrayList|Unknown
function EventSound:getStorySounds() end

---@public
---@return Color
function EventSound:getColor() end

---@public
---@return String
function EventSound:getName() end

---@public
---@return ArrayList|Unknown
function EventSound:getDataPoints() end

---@public
---@param arg0 Color
---@return void
function EventSound:setColor(arg0) end

---@public
---@param arg0 ArrayList|Unknown
---@return void
function EventSound:setDataPoints(arg0) end

---@public
---@param arg0 String
---@return void
function EventSound:setName(arg0) end

File: Vanilla/sources/zomboid/zombie/radio/StorySounds/EventSound.java
package zombie.radio.StorySounds;

import java.util.ArrayList;
import zombie.core.Color;

public final class EventSound {
   protected String name;
   protected Color color;
   protected ArrayList dataPoints;
   protected ArrayList storySounds;

   public EventSound() {
      this("Unnamed");
   }

   public EventSound(String var1) {
      this.color = new Color(1.0F, 1.0F, 1.0F);
      this.dataPoints = new ArrayList();
      this.storySounds = new ArrayList();
      this.name = var1;
   }

   public String getName() {
      return this.name;
   }

   public void setName(String var1) {
      this.name = var1;
   }

   public Color getColor() {
      return this.color;
   }

   public void setColor(Color var1) {
      this.color = var1;
   }

   public ArrayList getDataPoints() {
      return this.dataPoints;
   }

   public void setDataPoints(ArrayList var1) {
      this.dataPoints = var1;
   }

   public ArrayList getStorySounds() {
      return this.storySounds;
   }

   public void setStorySounds(ArrayList var1) {
      this.storySounds = var1;
   }
}
